sections/shared/modals/SelectTokenModal/SelectTokenModal.tsx

```diff
@@ -0,0 +1,242 @@
+	onDismiss: () => void;
+	onSelect: (currencyKey: CurrencyKey) => void;
+	tokensToOmit?: CurrencyKey[];
+};
+
+const ROW_HEIGHT = 43;
+
+export const SelectTokenModal: FC<SelectTokenModalProps> = ({
+	onDismiss,
+	onSelect,
+	tokensToOmit,
+}) => {
+	const { t } = useTranslation();
+	const [assetSearch, setAssetSearch] = useState<string>('');
+	const { selectPriceCurrencyRate, selectedPriceCurrency } = useSelectedPriceCurrency();
+
+	const tokenListQuery = use1InchTokenList();
+	const tokenList = useMemo(
+		() => (tokenListQuery.isSuccess ? tokenListQuery.data?.tokens ?? [] : []),
+		[tokenListQuery.isSuccess, tokenListQuery.data]
+	);
+
+	const tokensWalletBalancesQuery = useTokensBalancesQuery(tokenList);
+	const tokenBalances = tokensWalletBalancesQuery.isSuccess
+		? tokensWalletBalancesQuery.data ?? null
+		: null;
+
+	const isLoading = tokenListQuery.isLoading || tokensWalletBalancesQuery.isLoading;
+
+	const tokenBalancesAddresses = useMemo(
+		() =>
+			tokenBalances != null
+				? Object.values(tokenBalances).map((tokenBalance) => tokenBalance.token.address)
+				: [],
+		[tokenBalances]
+	);
+
+	const coinGeckoTokenPricesQuery = useCoinGeckoTokenPricesQuery(tokenBalancesAddresses);
+	const coinGeckoTokenPrices = coinGeckoTokenPricesQuery.isSuccess
+		? coinGeckoTokenPricesQuery.data ?? null
+		: null;
+
+	const coinGeckoPricesQuery = useCoinGeckoPricesQuery([CoinGeckoPriceIds.ETH]);
+	const coinGeckoPrices = coinGeckoPricesQuery.isSuccess ? coinGeckoPricesQuery.data ?? null : null;
+
+	const tokenBalancesWithPrices = useMemo(
+		() =>
+			tokenBalances != null
+				? Object.values(
+						mapValues(tokenBalances, ({ balance, token }, symbol) => {
+							const { address } = token;
+
+							const price =
+								symbol === CRYPTO_CURRENCY_MAP.ETH
+									? get(coinGeckoPrices, [CoinGeckoPriceIds.ETH, 'usd'], null)
+									: get(coinGeckoTokenPrices, [address.toLowerCase(), 'usd'], null);
+
+							return {
+								currencyKey: symbol,
+								balance,
+								usdBalance: price != null ? balance.multipliedBy(price) : null,
+								token,
+							};
+						})
+				  )
+				: [],
+		[coinGeckoPrices, coinGeckoTokenPrices, tokenBalances]
+	);
+
+	const searchFilteredTokens = useDebouncedMemo(
+		() =>
+			assetSearch
+				? tokenBalancesWithPrices.filter(({ token: { name, symbol, address } }) => {
+						const assetSearchQueryLC = assetSearch.toLowerCase();
+
+						return (
+							name.toLowerCase().includes(assetSearchQueryLC) ||
+							symbol.toLowerCase().includes(assetSearchQueryLC) ||
+							address.toLowerCase() === assetSearchQueryLC
+						);
+				  })
+				: tokenBalancesWithPrices,
+		[tokenBalancesWithPrices, assetSearch],
+		DEFAULT_SEARCH_DEBOUNCE_MS
+	);
+
+	const tokensResults = useMemo(() => {
+		const tokens = orderBy(
+			assetSearch ? searchFilteredTokens : tokenBalancesWithPrices,
+			(token) =>
+				token.usdBalance != null ? token.usdBalance.toNumber() : token.balance.toNumber(),
```

Comment:
altho its unlikely to not have `usdBalance`, its still better to surface them up using the `balance`